src/transaction/TransactionController.ts

```diff
@@ -649,17 +693,22 @@ export class TransactionController extends BaseController<
       )}`,
     );
 
-    const ethTransaction = new Transaction({
+    const txParams = {
       from: transactionMeta.transaction.from,
       gas: transactionMeta.transaction.gas,
```

Comment:
I think you should use gasLimit here instead of gas -- it may work still and I'm unsure why by TxData calls for 'gasLimit' https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/tx/src/types.ts#L82-L133

Reply:
this has now been adjusted.